options/ansi/generic/stdlib-stubs.cpp

```diff
@@ -421,10 +421,28 @@ size_t wcstombs(char *mb_string, const wchar_t *wc_string, size_t max_size) {
 	return wcsrtombs(mb_string, &wc_string, max_size, 0);
 }
 
+static bool debugMalloc() {
+	// There is a small delay between checking cached and setting debug,
+	// but this isn't a correctness issue so we ignore this for now.
+	// Note that we use uint32_t since some arches do not support 1 byte CAS.
+	static uint32_t cached = 0, debug = 0;
+
+	uint32_t expected = 1;
+	if (!__atomic_compare_exchange_n(&cached, &expected, true, false,
+				__ATOMIC_ACQUIRE, __ATOMIC_ACQUIRE)) {
+		// TODO: We should use a locking variant of getenv here to guard
+		// against concurrent modifications of the environment.
+		auto value = getenv("MLIBC_DEBUG_MALLOC") != nullptr;
+		__atomic_store_n(&debug, value, __ATOMIC_RELAXED);
+		return value;
+	}
+
+	return __atomic_load_n(&debug, __ATOMIC_RELAXED);
+}
```

Comment:
I think this can be vastly simplified to:
```suggestion
static bool debugMalloc() {
	static bool debug = [] () -> bool {
		return getenv("MLIBC_DEBUG_MALLOC");
	}(); // Immediately invoked.
	return debug;
}
```
while also avoiding the race. Note that it would be a bit nicer to also check the value of `MLIBC_DEBUG_MALLOC` (right now, `MLIBC_DEBUG_MALLOC=0` turns on debugging, which is a bit counterintuitive).